### PR TITLE
dp/invoiceitems-relationships

### PR DESCRIPTION
**Pull request by Dpalazzari**
In this PR:

- Invoice items with invoice relationship API functionality and test
- invoice items with item relationship API functionality and test

Take a look! @sallymacnicholas @tmikeschu

### CODE REVIEW

File: app/controllers/api/v1/invoice_items/invoice_items_inv_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Api::V1::InvoiceItems::InvoiceItemsInvController < ApplicationController
```
Comment (tmikeschu) [inline app/controllers/api/v1/invoice_items/invoice_items_inv_controller.rb]: Can we fully name Inv to Invoice? I can see the point of abbreviating, but it's only four extra characters.

File: config/routes.rb [hunk 2]
```diff
@@ -19,6 +19,8 @@
         get '/find' => 'search#show'
         get '/find_all' => 'search#index'
         get '/random' => 'random#show'
+        get ':id/invoice', to: 'invoice_items_inv#show'
```
Comment (tmikeschu) [inline config/routes.rb]: Getting those custom shows! Love the OO design.

**Review by tmikeschu — COMMENTED** [review]
@Dpalazzari great work! Just one comment about the invoice item invoice controller name

**Comment by Dpalazzari** [comment]
@tmikeschu Refactored!

**Pull request merged**